pod/perldata.pod

```diff
@@ -520,13 +522,95 @@ C<$who::0>, and a C<$who's> variable.  The last two would be the
 $0 and the $s variables in the (presumably) non-existent package
 C<who>.
 
-In fact, a simple identifier within such curlies is forced to be
-a string, and likewise within a hash subscript.  Neither need
-quoting.  Our earlier example, C<$days{'Feb'}> can be written as
-C<$days{Feb}> and the quotes will be assumed automatically.  But
-anything more complicated in the subscript will be interpreted as an
-expression.  This means for example that C<$version{2.0}++> is
-equivalent to C<$version{2}++>, not to C<$version{'2.0'}++>.
+In fact, a simple identifier within such curly braces is forced to be a
+string, and likewise within a hash subscript. Neither need quoting. Our
+earlier example, C<$days{'Feb'}> can be written as C<$days{Feb}> and the
+quotes will be assumed automatically. But anything more complicated in
+the subscript will be interpreted as an expression. This means for
+example that C<$version{2.0}++> is equivalent to C<$version{2}++>, not
+to C<$version{'2.0'}++>.
+
+There is a similar problem with interpolation with text that looks like
+array or hash access notation. Placing a simple variable like C<$who>
+immediately in front of text like C<"[1]"> or C<"{foo}"> would cause the
+variable to be interpolated as accessing an element of C<@who> or a
+value stored in C<%who>:
+
+    $who = "Larry Wall";
+    print "$who[1] is the father of the Perl programming language\n";
+
+would attempt to access index 1 of an array named C<@who>. Again, using
+braces will prevent this from happening:
+
+    $who = "Larry Wall";
+    print "${who}[1] is the father of the Perl programming language\n";
+
+will be treated the same as
+
+    $who = "Larry Wall";
+    print $who, "[1] is the father of the Perl programming language\n";
```

Comment:
Shouldn't this be `print $who . "[1] is the father of ...`?
With commas `print` will receive a list of strings, which is not the same as variable interpolations.

Reply:
Yes, of course you are right! I was thinking purely in the sense of the side-effect, but in the docs we should be more specific! Thanks. Fixed, and pushed!